linalg/sparsemat.cpp

```diff
@@ -121,16 +121,17 @@ SparseMatrix::SparseMatrix(int nrows, int ncols, int rowsize)
    }
 }
 
-SparseMatrix::SparseMatrix(const SparseMatrix &mat, bool copy_graph)
+SparseMatrix::SparseMatrix(const SparseMatrix &mat, bool copy_graph,
+                           MemoryType mt)
    : AbstractSparseMatrix(mat.Height(), mat.Width())
 {
    if (mat.Finalized())
    {
       const int nnz = mat.I[height];
       if (copy_graph)
       {
-         I.New(height+1, mat.I.GetMemoryType());
-         J.New(nnz, mat.J.GetMemoryType());
+         I.New(height+1, mt == MemoryType::SIZE ? mat.I.GetMemoryType() : mt);
```

Comment:
I'm not sure we should use `MemoryType::SIZE` here, it's confusing.
Do we need to add a `UNKNOWN` one?

Reply:
I agree it would be good to have something else. We could call it `UNSET`, which would make the intention clear when it was used as the default arg but I also like `UNKNOWN` and `DEFAULT`. Would you like to hold this PR until we add one?

Reply:
Tom, does this need to be in the Copy constructor, e.g. can we add another constructor that explicitly provides the memory type?

Reply:
It does need to be in the copy constructor. I think we should actually move these changes to GH-1644 since the interface might change and that's where we're working on it

Reply:
I'm Okay with that. Do you still want to keep this PR open?

Reply:
No, let's close it

Reply:
thanks Tom!